R/arx_forecaster.R

```diff
@@ -119,22 +123,62 @@ arx_fcast_epi_workflow <- function(
   if (!(is.null(trainer) || is_regression(trainer))) {
     cli_abort("`trainer` must be a {.pkg parsnip} model of mode 'regression'.")
   }
+  # forecast_date is above all what they set;
+  # if they don't and they're not adjusting latency, it defaults to the max time_value
+  # if they're adjusting, it defaults to the as_of
+  if (args_list$adjust_latency == "none") {
+    forecast_date_default <- max(epi_data$time_value)
+    if (!is.null(args_list$forecast_date) && args_list$forecast_date != forecast_date_default) {
+      cli_warn(
+        "The specified forecast date {args_list$forecast_date} doesn't match the date from which the forecast is actually occurring {forecast_date_default}.",
+        class = "epipredict__arx_forecaster__forecast_date_defaulting"
+      )
+    }
+  } else {
+    forecast_date_default <- attributes(epi_data)$metadata$as_of
+  }
+  forecast_date <- args_list$forecast_date %||% forecast_date_default
+  target_date <- args_list$target_date %||% (forecast_date + args_list$ahead)
+  if (forecast_date + args_list$ahead != target_date) {
+    cli_abort(
+      c(
+        "`forecast_date` + `ahead` must equal `target_date`.",
+        i = "{.val {forecast_date}} + {.val {ahead}} != {.val {target_date}}."
+      ),
+      class = "epipredict__arx_forecaster__inconsistent_target_ahead_forecaste_date"
+    )
+  }
```

Comment:
Something like this is my preference if it passes styler. The `c(` on its own line kinda bugs me.

```suggestion
    cli_abort(
        "`forecast_date` {.val {forecast_date}} + `ahead` {.val {ahead}} must equal `target_date` {.val {target_date}}.",
      class = "epipredict__arx_forecaster__inconsistent_target_ahead_forecaste_date"
    )
  }
```